api4/team.go

```diff
@@ -808,7 +817,15 @@ func searchTeams(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	c.App.SanitizeTeams(c.App.Session, teams)
 
-	w.Write([]byte(model.TeamListToJson(teams)))
+	var payload []byte
+	if props.Paginate {
+		twc := &model.TeamsWithCount{Teams: teams, TotalCount: totalCount}
+		payload = model.TeamsWithCountToJson(twc)
+	} else {
+		payload = []byte(model.TeamListToJson(teams))
```

Comment:
It would be nice if both of these versions returned the same type of data, even if the total count is a bit redundant when you're not using paging. That'll be something to consider for the next major API version.

Reply:
Totally agreed that it would be nice for the structure to be the same and to be done on a major version bump. I'm going to add a story for this.